data/src/main/java/com/microsoft/azure/kusto/data/ConnectionStringBuilder.java

```diff
@@ -142,32 +115,32 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
         return csb;
     }
 
-    public static ConnectionStringBuilder createWithAadApplicationCredentials(String resourceUri,
-                                                                              String applicationClientId,
-                                                                              String applicationKey) {
-        return createWithAadApplicationCredentials(resourceUri, applicationClientId, applicationKey, null);
+    public static ConnectionStringBuilder createWithUserPrompt(String resourceUri) {
+        return createWithUserPrompt(resourceUri, DEFAULT_DEVICE_AUTH_TENANT, null);
+    }
+
+    public static ConnectionStringBuilder createWithUserPrompt(String resourceUri, String userName) {
+        return createWithUserPrompt(resourceUri, DEFAULT_DEVICE_AUTH_TENANT, userName);
     }
```

Comment:
i dont know if you tried it but its so much faster then the adal device auth :)

Reply:
Yeah it was fast and smooth as butter!